src/common/src/error.rs

```diff
@@ -191,17 +189,8 @@ impl From<RwError> for tonic::Status {
             ErrorCode::OK => tonic::Status::ok(err.to_string()),
             ErrorCode::ExprError(e) => tonic::Status::invalid_argument(e.to_string()),
             ErrorCode::PermissionDenied(e) => tonic::Status::permission_denied(e),
-            _ => {
-                let bytes = {
-                    let status = err.to_status();
-                    let mut bytes = Vec::<u8>::with_capacity(status.encoded_len());
-                    status.encode(&mut bytes).expect("Failed to encode status.");
-                    bytes
-                };
-                let mut header = MetadataMap::new();
-                header.insert_bin(RW_ERROR_GRPC_HEADER, MetadataValue::from_bytes(&bytes));
-                tonic::Status::with_metadata(Code::Internal, err.to_string(), header)
-            }
```

Comment:
(BTW, I just removed `#![expect(dead_code)]`, so after merging main clippy will fail 🥸

Reply:
```
impl RwError {
    /// Converting to risingwave's status.
    ///
    /// We can't use grpc/tonic's library directly because we need to customized error code and
    /// information.
    fn to_status(&self) -> Status {
        // TODO: We need better error reporting for stacktrace.
        Status {
            code: self.inner.get_code() as i32,
            message: self.to_string(), /// Simply discard the details and encode itself to string.
        }
    }
```

Actually, we didn't encode the enum `ErrorCode` into a protobuf struct and decode it completely as it was, instead, the ErrorCode will always be an `InternalError` containing a string, which was encoded from the original code. 

Hence, a ConnectorError, after being transported via grpc, will become:

```
internal error: connector error: xxxx
```

Reply:
Yes, the current implementation doesn't make full sense, but the idea might be good in the future. Anyway I think it needs to be carefully re-designed, so I think it's also ok to remove it and KISS for now.